bson/encode.go

```diff
@@ -661,11 +669,18 @@ func (e *encoder) valueFromValue(val reflect.Value, minsize bool) (*Value, error
 			elem = VC.ArrayFromValues(arrayElems...)
 		}
 	case reflect.Struct:
-		structElems, err := e.encodeStruct(val)
-		if err != nil {
-			return nil, err
+		switch val.Interface().(type) {
+		case time.Time:
```

Comment:
There's currently no test covering this. This can be covered by adding a `[]time.Time` case to the `struct{[]interface{}}` test.